scala-thoughts.md

```diff
@@ -0,0 +1,15 @@
+
+#### What is/are your least favorite language design choice(s)? Why? And why do you think the designers made that / those choice(s)?
+
+My least favorite language design choice was the amount of built-in functionality. It definitely felt extremely overwhelming when Eclipse shows all the built-in methods. Again, I feel like this experience is unique to someone new to Scala, and probably doesn't reflect someone well-versed. For me, it was frustrating knowing that there existed a built-in function that did exactly what I needed, but just not knowing which of the hundreds available. It reminded me a bit of what Stone was saying in his article about how users could get overwhelmed if the language was too large. I was also surprised how scattered some documentation was for less frequently used built-in functions. For instance, I remember trying to find some documentation from Scala for foldLeft, but could only find third party sites that had extremely verbose explanations. Stackoverflow definitely saved me a lot of time in terms of these built-in functions by providing concise explanations with effective examples. It makes sense that the designers implemented these built-in functions as I would imagine a language designer using Scala would want as many tools as possible. 

```

Comment:
Something I did at first was tend to assume that built-ins are universal like in Python, but as I think you've seen by now, if it makes sense for a built-in to apply to a class, it's a method of that class. Did you look at e.g. [this](http://www.scala-lang.org/api/2.11.7/index.html#scala.collection.immutable.List@foldLeft[B](z:B)(f:(B,A)=>B):B) and find it unhelpful? If not I highly recommend it for future reference, I found this source of documentation quite helpful during this assignment.